framework/aster-frame/src/vm/config.rs

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! A configuration module for framework memory management.
+
+use super::Vaddr;
+
+pub const KERNEL_STACK_SIZE: usize = PAGE_SIZE * 64;
+pub const KERNEL_HEAP_SIZE: usize = PAGE_SIZE * 256;
+
+/// Typicall 64-bit systems have at least 48-bit virtual address space.
+/// Halving it to reserve half of the address space for the kernel.
+/// 
+/// Also, the top page is not regarded as usable since it's a workaround
+/// for the x86_64 CPUs' bugs. See
+/// <https://github.com/torvalds/linux/blob/480e035fc4c714fb5536e64ab9db04fedc89e910/arch/x86/include/asm/page_64.h#L68-L78>
+/// for the rationale.
+pub const USERSPACE_LOWEST_UNUSABLE_VADDR: Vaddr = 0x0000_8000_0000_0000 - PAGE_SIZE;
```

Comment:
The const name is a bit of confusing becauses it only states no address below is unusable. It is not clear if some addresses above it are usable or will be used. To make it more clear, I suggest calling it `MAX_USER_VADDR`.